fem/libceed/mf_integrator.hpp

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_LIBCEED_MFINTEG
+#define MFEM_LIBCEED_MFINTEG
+
+#include "operator.hpp"
+#include "../../config/config.hpp"
+#include "coefficient.hpp"
+#include "../fespace.hpp"
+#include "../gridfunc.hpp"
+
+namespace mfem
+{
+
+namespace ceed
+{
+
+#ifdef MFEM_USE_CEED
+/** This structure contains the data to assemble a MF operator with libCEED.
+    See libceed/mass.cpp or libceed/diffusion.cpp for examples. */
+struct MFOperator
+{
+   /** The finite element space for the trial and test functions. */
+   const mfem::FiniteElementSpace &fes;
+   /** The Integration Rule to use to compote the operator. */
+   const mfem::IntegrationRule &ir;
+   /** The path to the header containing the functions for libCEED. */
+   std::string header;
+   /** The name of the Qfunction to apply the operator. */
+   std::string apply_func;
+   /** The Qfunction to apply the operator. */
+   CeedQFunctionUser apply_qf;
+   /** The evaluation mode to apply to the trial function (CEED_EVAL_INTERP,
+       CEED_EVAL_GRAD, etc.) */
+   EvalMode trial_op;
+   /** The evaluation mode to apply to the test function ( CEED_EVAL_INTERP,
+       CEED_EVAL_GRAD, etc.) */
+   EvalMode test_op;
+};
+#endif
+
+class MFIntegrator : public Operator
```

Comment:
I think this class needs to be documented, especially since `PAIntegrator` inherits from it.

Reply:
I added a short documentation, and `PAIntegrator` no longer inherit from `MFIntegrator`.